### PR TITLE
#9103 No update button for pinned deployments

### LINKED ISSUE #9103: Mods screen shows "update" button for team deployments
## Describe the bug

- Mods screen shows "update" button for deployments
- The intent of deployments is that users will use the pinned version
- The button is confusing because users naturally think they're supposed to click the button

## To Reproduce

1. Create a deployment of a mod and pin to an older version
2. Deploy the mod via deployment key (or normally to a restricted user)

## Actual behavior

The mods screen shows an "Update" button for the mod

## Expected behavior

- The mods screen shows an "Active" status with the pinned mod version for the mod

## Discussion

- For organizations that apply UI restrictions, the [RestrictedFeatures.DEACTIVATE_DEPLOYMENT](https://github.com/pixiebrix/pixiebrix-extension/blob/a0e8129cf370b17a2b2505ba772cd65e4e1242b0/src/auth/featureFlags.ts#L156-L156) would be the policy for whether or not the use is able to deactivate/re-activate a deployment
- However, to avoid confusion in the UI, for deployments, the mods screen SHOULD NOT show the Update button as the main status to any user. But users with permission to reactivate should still be able to re-activate/upgrade using the 3-dot menu to get the latest version
- Note that the Update button should show for personal deployments: https://github.com/pixiebrix/pixiebrix-extension/issues/9091

## Future Work

- Indicate to developers/managers/admins when a mod deployment isn't the latest version (and see which version is activated vs. the latest)

## Related Code

- https://github.com/pixiebrix/pixiebrix-extension/blob/a0e8129cf370b17a2b2505ba772cd65e4e1242b0/src/extensionConsole/pages/mods/Status.tsx#L91-L91
- https://github.com/pixiebrix/pixiebrix-extension/blob/a0e8129cf370b17a2b2505ba772cd65e4e1242b0/src/extensionConsole/pages/mods/ModsPageActions.tsx#L167-L167

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #9103
- Also refactors hasUpdate logic in buildGetModVersionStatus

## Demo

https://www.loom.com/share/a16175dd5bb945309f325b110ef78559

### CODE REVIEW

File: src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts [hunk 1]
```diff
@@ -17,73 +17,97 @@
...
+  if (current == null || activated == null) {
+    return true; // No update available if either version is null
+  }
+
+  if (semver.gt(current.version, activated.version)) {
+    return false;
+  }
+
+  if (semver.lt(current.version, activated.version)) {
+    // TODO: In what case would the current version be less than the activated version?
```
Comment (mnholtz) [inline src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts:44]: When refactoring, this logic came to my attention 🤔  Anyone know what this logic could be handling?

File: src/extensionConsole/pages/mods/Status.tsx [hunk 2]
```diff
@@ -88,7 +89,7 @@ const Status: React.VoidFunctionComponent<{
     );
   }
 
-  if (hasUpdate && showReactivate) {
+  if (hasUpdate && showReactivate && !isDeployment) {
```
Comment (mnholtz) [inline src/extensionConsole/pages/mods/Status.tsx:92]: This is the change for the majority of the feature, aside from the personal deployment case

File: src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts
Comment (mnholtz) [inline src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts:1]: Unrelated refactor (I initially though my logic would go here)

File: src/extensionConsole/pages/mods/Status.tsx [hunk 3]
```diff
@@ -88,7 +90,7 @@ const Status: React.VoidFunctionComponent<{
     );
   }
 
-  if (hasUpdate && showReactivate) {
+  if (hasUpdate && showReactivate && !(sharingSource.type === "Deployment")) {
```
Comment (mnholtz) [inline src/extensionConsole/pages/mods/Status.tsx:93]: This is the fix

File: src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts (hunk 1 above)
Comment (grahamlangford) [reply src/extensionConsole/pages/mods/utils/buildGetModVersionStatus.ts:44]: Probably no case, just completionist

File: src/extensionConsole/pages/mods/Status.tsx [hunk 4]
```diff
@@ -88,11 +90,12 @@ const Status: React.VoidFunctionComponent<{
...
     return (
       <Button
         size="sm"
         variant="info"
+        className="text-nowrap"
```
Comment (mnholtz) [inline src/extensionConsole/pages/mods/Status.tsx:98]: Unrelated fix to the wrapping issue on the Update button
Screenshot 2024-09-12 at 11 22 37 AM

File: src/extensionConsole/pages/mods/Status.test.tsx [hunk 5]
```diff
@@ -87,6 +87,44 @@ describe("Status", () => {
     expect(screen.getByText("Update")).toBeInTheDocument();
   });
 
+  it("doesn't show update button for deployments", () => {
```
Comment (fungairino) [inline src/extensionConsole/pages/mods/Status.test.tsx:90]: nit: I'd be more specific and name this test "doesn't show update button for team deployments"

**Review by fungairino — APPROVED** [review]

**Pull request merged**